lib/bundler/definition.rb

```diff
@@ -409,8 +409,10 @@ def ensure_equivalent_gemfile_and_lockfile(explicit_flag = false)
              "updated #{Bundler.default_lockfile.relative_path_from(SharedHelpers.pwd)} to version control."
 
       unless explicit_flag
+
+        suggest_command = Bundler.settings.locations("frozen")[:global] == "1" ? "bundle config --delete frozen" : "bundle install --no-deployment"
```

Comment:
May be this is expected?

`➜  bundler-dev cat ~/.bundle/config
---
BUNDLE_DISABLE_CHECKSUM_VALIDATION: "true"
BUNDLE_FROZEN: "1"
➜  bundler-dev cat .bundle/config
---
BUNDLE_PATH: "vendor/bundle"
BUNDLE_DISABLE_SHARED_GEMS: "true"
BUNDLE_FROZEN: "1"
➜  bundler-dev dbundle config --delete frozen
➜  bundler-dev cat ~/.bundle/config
---
BUNDLE_DISABLE_CHECKSUM_VALIDATION: "true"
➜  bundler-dev cat .bundle/config
---
BUNDLE_PATH: "vendor/bundle"
BUNDLE_DISABLE_SHARED_GEMS: "true"
➜  bundler-dev`